### ISSUE TITLE
Way to retrieve value of properties in `pipe`

### ISSUE THREAD

**Issue by grst** [machow/siuba#324]
To execute a method of an object, I can do something like: 

```python
# Executes the `describe` method of `mtcars`
mtcars >> pipe(_.describe())
```

However, when I want to retrieve a property of an object, this does not work:
```python
# Want to get `mtcars.columns`
mtcars >> pipe(_.columns)
```
```console
█─'__call__'
├─█─.
│ ├─_
│ └─'columns'
└─     mpg  cyl   disp   hp  drat     wt   qsec  vs  am  gear  carb
  0   21.0    6  160.0  110  3.90  2.620  16.46   0   1     4     4
  1   21.0    6  160.0  110  3.90  2.875  17.02   0   1     4     4
  2   22.8    4  108.0   93  3.85  2.320  18.61   1   1     4     1
[...]
```

The closest I could get was: 
```python
mtcars >> pipe(lambda _: _.columns)
```
```console
Index(['mpg', 'cyl', 'disp', 'hp', 'drat', 'wt', 'qsec', 'vs', 'am', 'gear',
       'carb'],
      dtype='object')
```

But it would be really cool to retrieve properties directly from the `Symbolic` instance.

**Comment by grst** [comment]
Hi @machow, 

I gave the PR a try and the basic functionality seems to work. I am still having issues with a more complex object, though. 
The minimal example may seem a bit pathological, but I believe this should work. My real use-case is to manipulate a [`AnnData`](https://anndata.readthedocs.io/) object (a container for gene expression data based on pandas) in a siuba pipeline. 

```python
class DataframeContainer:
    def __init__(self, df):
        self.df = df

dfc = DataframeContainer(mtcars)

# works
dfc >> pipe(lambda _ : _.df.loc[_.df["mpg"] > 20, :])

# AttributeError
dfc >> pipe(_.df.loc[_.df["mpg"] > 20, :])
```

<details>
<summary>Traceback</summary>
```pytb
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-46-ef386b97c605> in <module>
----> 1 dfc >> pipe(_.df.loc[_.df["mpg"] > 20, :])

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/siuba/dply/verbs.py in __rrshift__(self, x)
     97             return Pipeable(calls = [x] + self.calls)
     98 
---> 99         return self(x)
    100 
    101     def __call__(self, x):

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/siuba/dply/verbs.py in __call__(self, x)
    102         res = x
    103         for f in self.calls:
--> 104             res = f(res)
    105         return res
    106 

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/siuba/siu.py in __call__(self, x)
    197         # in normal case, get method to call, and then call it
    198         f_op = getattr(operator, self.func)
--> 199         return f_op(inst, *rest, **kwargs)
    200 
    201     @staticmethod

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/pandas/core/indexing.py in __getitem__(self, key)
    882     def __getitem__(self, key):
    883         if type(key) is tuple:
--> 884             key = tuple(com.apply_if_callable(x, self.obj) for x in key)
    885             if self._is_scalar_access(key):
    886                 with suppress(KeyError, IndexError, AttributeError):

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/pandas/core/indexing.py in <genexpr>(.0)
    882     def __getitem__(self, key):
    883         if type(key) is tuple:
--> 884             key = tuple(com.apply_if_callable(x, self.obj) for x in key)
    885             if self._is_scalar_access(key):
    886                 with suppress(KeyError, IndexError, AttributeError):

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/pandas/core/common.py in apply_if_callable(maybe_callable, obj, **kwargs)
    327     """
    328     if callable(maybe_callable):
--> 329         return maybe_callable(obj, **kwargs)
    330 
    331     return maybe_callable

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/siuba/siu.py in __call__(self, *args, **kwargs)
    678     def __call__(self, *args, **kwargs):
    679         if self.__ready_to_call:
--> 680             return self.__source(*args, **kwargs)
    681 
    682         return create_sym_call(self.__source, *args, **kwargs)

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/siuba/siu.py in __call__(self, x)
    186 
    187     def __call__(self, x):
--> 188         inst, *rest = (self.evaluate_calls(arg, x) for arg in self.args)
    189         kwargs = {k: self.evaluate_calls(v, x) for k, v in self.kwargs.items()}
    190 

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/siuba/siu.py in <genexpr>(.0)
    186 
    187     def __call__(self, x):
--> 188         inst, *rest = (self.evaluate_calls(arg, x) for arg in self.args)
    189         kwargs = {k: self.evaluate_calls(v, x) for k, v in self.kwargs.items()}
    190 

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/siuba/siu.py in evaluate_calls(arg, x)
    201     @staticmethod
    202     def evaluate_calls(arg, x):
--> 203         if isinstance(arg, Call): return arg(x)
    204 
    205         return arg

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/siuba/siu.py in __call__(self, x)
    186 
    187     def __call__(self, x):
--> 188         inst, *rest = (self.evaluate_calls(arg, x) for arg in self.args)
    189         kwargs = {k: self.evaluate_calls(v, x) for k, v in self.kwargs.items()}
    190 

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/siuba/siu.py in <genexpr>(.0)
    186 
    187     def __call__(self, x):
--> 188         inst, *rest = (self.evaluate_calls(arg, x) for arg in self.args)
    189         kwargs = {k: self.evaluate_calls(v, x) for k, v in self.kwargs.items()}
    190 

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/siuba/siu.py in evaluate_calls(arg, x)
    201     @staticmethod
    202     def evaluate_calls(arg, x):
--> 203         if isinstance(arg, Call): return arg(x)
    204 
    205         return arg

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/siuba/siu.py in __call__(self, x)
    191         # TODO: temporary workaround, for when only __get_attribute__ is defined
    192         if self.func == "__getattr__":
--> 193             return getattr(inst, *rest)
    194         elif self.func == "__call__":
    195             return getattr(inst, self.func)(*rest, **kwargs)

~/anaconda3/envs/test_siuba/lib/python3.8/site-packages/pandas/core/generic.py in __getattr__(self, name)
   5463             if self._info_axis._can_hold_identifiers_and_holds_name(name):
   5464                 return self[name]
-> 5465             return object.__getattribute__(self, name)
   5466 
   5467     def __setattr__(self, name: str, value) -> None:

AttributeError: 'DataFrame' object has no attribute 'df'
```

</details>

**Comment by machow** [comment]
Ah, thanks for the example--it looks like there's a bug in how siuba represents slices with multiple entries.

Right now, siu expressions handle slices by seeing whether the slice is another siu expressions:

```python
_[_ > 20]          # knows _ > 20 is symbolic
```

```
# output
█─[
├─_
└─█─>
  ├─_
  └─20
```

However, multiple entries to `[` produce a tuple of slices, which it is skipping over..

```python
_[_ > 20, :] 
```

```
█─[
├─_
└─(█─>            # notice the ( means all this is in a tuple
  ├─_
  └─20, slice(None, None, None))
```

**Comment by machow** [comment]
Alright--this ended up being a much bigger fix, but should be working now on the PR branch! I'm going to clean up the code a bit, and will try to merge over the next couple days. Thanks again for the example!

**Comment by grst** [comment]
I can confirm this works :tada: 
Thanks a lot for the quick fix!

**Comment by machow** [comment]
Sorry for the wait--should be available in v0.0.25!
